leader-election-impl/src/main/java/com/palantir/paxos/PaxosQuorumChecker.java

```diff
@@ -189,8 +189,15 @@ public RESPONSE call() throws Exception {
 
         } finally {
             // cancel pending futures (during failures)
+            boolean canceled = false;
             for (Future<RESPONSE> future : allFutures) {
-                future.cancel(false);
+                if (!future.isDone()) {
+                    future.cancel(true);
+                    canceled = true;
+                }
```

Comment:
We might get some false positives this way I think, if the future completes between the check on `isDone()` and `cancel()` actually taking place. I _think_ it's probably enough to write

```
if (future.cancel(true)) { // safely returns false if the task has already completed
  canceled = true;
}
```

Reply:
oh! right, forgot that works.